chtest.c

```diff
@@ -0,0 +1,235 @@
+   Driver_Class = class_create(THIS_MODULE, CLASS_NAME);
+   if (IS_ERR(Driver_Class)){                // Check for error and clean up if there is
+      unregister_chrdev(majorNumber, DEVICE_NAME);
+      printk(KERN_ALERT "Failed to register device class\n");
+      return PTR_ERR(Driver_Class);          // Correct way to return an error on a pointer
+   }
+   printk(KERN_INFO "chtest: device class registered correctly\n");
+ 
+   // Register the device driver
+   Driver_Device = device_create(Driver_Class, NULL, MKDEV(majorNumber, 0), NULL, DEVICE_NAME);
+   if (IS_ERR(Driver_Device)){               // Clean up if there is an error
+      class_destroy(Driver_Class);           // Repeated code but the alternative is goto statements
+      unregister_chrdev(majorNumber, DEVICE_NAME);
+      printk(KERN_ALERT "chtest Failed to create the device\n");
+      return PTR_ERR(Driver_Device);
+   }
+   printk(KERN_INFO "chtest: device class created correctly\n"); // Made it! device was initialized
+   procptr=proc_create("chtest",0666,NULL,&procops);       // proc create
+   printk(KERN_INFO "chtest  proc entry created successfully");	
+
+chtest_kobject = kobject_create_and_add("chtestflag",kernel_kobj); //sys
+if(!chtest_kobject) //sys init
+                return -ENOMEM;
+
+       int error = sysfs_create_file(chtest_kobject, &sys_attribute.attr);
+
+        if (error) {
+               printk(KERN_ALERT "failed to create the clrflag file in /sys/kernel/chtest_kobject\n");
+}
+   return 0;
+
+}
+
+					//proc
+static ssize_t procwrite(struct file *file, const char __user *ubuf,size_t count, loff_t *ppos) //proc write func 
+{
+	printk( KERN_DEBUG "procwrite handler\n");
+	return -1;
+}
+
+ 
+static ssize_t procread(struct file *file, char __user *ubuf,size_t count, loff_t *ppos) //proc read func 
+{
+	int BUFSIZE = 100;
+	char buf[BUFSIZE];
+	int len=0;
+	if(*ppos > 0 || count < BUFSIZE)
+		return 0;
+	if(B_size > 0) {
+	len += sprintf(buf,"BUFFER_SIZE = %d\n",B_size );
+	len += sprintf(buf + len,"USED_BUFFER = %d",size_of_message );
+	}
+else {
+len += sprintf(buf,"BUFFER_SIZE = %d\n",BSIZE );
+	len += sprintf(buf + len,"USED_BUFFER = %d\n",size_of_message);
+}
+	if(copy_to_user(ubuf,buf,len))
+		return -EFAULT;
+	*ppos = len;
+	return len;
+}
+ 
+static void proc_cleanup(void)    //proc cleanup func
+{
+	remove_proc_entry("chtest",NULL);
+}
+
+ 			//Sys
+static ssize_t clrflag_show(struct kobject *kobj, struct kobj_attribute *chtest_kobject, char *buf)
+{
+
+        return sprintf(buf, "%d\n", clrflag);
+}
+
+static ssize_t clrflag_store(struct kobject *kobj, struct kobj_attribute *chtest_kobject,  char *buf, size_t count)
+{
+        sscanf(buf, "%du", &clrflag);
+
+if (clrflag == 1) {
+if(B_size > 0) {
+memset(new_buff,0,B_size);
+}
+else memset(&message[0],0,BSIZE);
+
+}
+size_of_message =0;
+        return count;							
+}
+
+static void sys_cleanup(void)
+{
+
+	kobject_put(chtest_kobject);
+}
+
+static void __exit chtest_exit(void){
```

Comment:
```suggestion
static void __exit chtest_exit(void)
{
```
Please fix Kernel code style